folly/io/async/AsyncSSLSocket.h

```diff
@@ -175,9 +175,9 @@ class AsyncSSLSocket : public AsyncSocket {
 
     void readDataAvailable(size_t len) noexcept override {
       CHECK_EQ(len, 1);
-      sslSocket_->restartSSLAccept();
       pipeReader_->setReadCB(nullptr);
       sslSocket_->setAsyncOperationFinishCallback(nullptr);
+      sslSocket_->restartSSLAccept();
```

Comment:
Please add a comment here explaining why order is significant

Reply:
@afrind  I have addressed all your comments, please review